packages/nvidia_nat_config_optimizer/src/nat/plugins/config_optimizer/prompts/ga_prompt_optimizer.py

```
@@ -0,0 +1,697 @@
+                                param,
+                                e,
+                            )
+                    child_prompts[param] = child
+                return _make_individual(child_prompts)
+
+            def _select_parent(curr_pop: list[Individual]) -> Individual:
+                if selection_method == "tournament":
+                    return self._tournament_select(curr_pop, tournament_size)
+                elif selection_method == "roulette":
+                    total = (sum(max(ind.scalar_fitness or 0.0, 0.0) for ind in curr_pop) or 1.0)
+                    r = random.random() * total
```

Ruff (0.15.5)

[error] Standard pseudo-random generators are not suitable for cryptographic purposes

(S311)